protocol/deprecate-reqresp-sync.md

```diff
@@ -0,0 +1,138 @@
+we had to jump through a few hoops, to handle stateless genesis for OP Mainnet,
+which has a non-zero bedrock-migration "genesis" state.
+
+[Mid 2023](https://github.com/ethereum-optimism/op-geth/pull/105) Sunnyside Labs then
+owned Snap-sync development for a while, improving the usability.
+
+### Introduction of SyncMode
+
+[End 2023](https://github.com/ethereum-optimism/optimism/pull/8333) `trianglesphere` then improved the
+integration again, by introducing a "sync-mode" configuration to op-node.
+
+### P2P req-resp refresh attempt
+
+[July 2024] New help with P2P was meant to onboard
+[with context](https://www.notion.so/oplabs/op-node-P2P-context-b806628dd29c45acbe69dac2b02c63fb)
+Followed by [a P2P req-resp rewrite](https://github.com/ethereum-optimism/optimism/pull/11781) that was never merged
+due to various setbacks.
+
+## Retrospective
+
+- The stack started with block gossip without EL sync
+- Then introduced snap-sync and introduced different sync-modes
+- Then effectively reached a complexity-ceiling when more changes were needed
+
+# Proposed Solution
+
+Less code is better code.
+The EL node has a specialized sync implementation.
+
+Specifically, EL nodes can sync to the tip of the chain, when connected to healthy nodes,
+and when given a full target block-header to sync to.
+
+`op-geth` does so by maintaining a DB of synced blocks (i.e. memory is not a limit to syncing),
+and maintaining a "skeleton chain" of block-headers to connect between the known chain and the target.
+This also handles a moving target, and reorgs during sync.
+
+### Ensuring a sync-target
+
+With gossip-sub, target block-headers are generally available.
+
+The main exception is when a node restarts, and when gossip-sub is idle (e.g. sequencer is offline),
+then nodes might be unable to get a new sync-target.
+
+We can solve this by implementing the following change:
+- Cache the last gossiped block with signature.
+- Change the gossip-relay rules:
+  accept older blocks that don't meet the wallclock validation rules, but only up to a certain rate-limit.
+  Whenever an older block is received than the cached last block,
+  then gossipsub-`IGNORE` the older block, and publish the cached block.
+  This prevents stale blocks from being replayed without bound,
+  as the cached block will be replayed instead, which more nodes can agree on and filter out.
+- (Maybe) introduce an extremely simple request-response protocol,
+  "poke", that requests a node to share its last signed block.
+  When idle, or just connecting, this can help start sync.
+
+### Simplifications
+
+To roll out, we need:
+- Review sync errors / state returned by the engine API,
+  and clarify in op-node metrics/logs what the sync-state is.
+- Make EL-sync the default. We can keep CL-sync, but e.g. add a deprecation warning.
+- Sync-test that we can rely solely on EL-sync.
+- Update specs to deprecate the request-response block sync.
+- Update docs to remove CL/EL-sync differentiation.
```

Comment:
Maybe we want to clarify that L1 CL sync is still running for deriving the _safe_ chain, and we'd default to EL-sync for _unsafe_ chain sync only?

Reply:
will clarify here as well